src/inference/hmc.jl

```diff
@@ -178,13 +178,22 @@ function DynamicPPL.initialstep(
     # If no initial parameters are provided, resample until the log probability
     # and its gradient are finite.
     if init_params === nothing
+        init_attempt_count = 0
+        init_user_warned = false
```

Comment:
Why is this needed? It seems it would be equivalent to only check `init_attempt_count == 10` below?
```suggestion
```